File: fib-iterate.py
import sys
from time import time
def fib(n):
	a,b = 1,1
	for i in range(3,n+1):
		c = a + b
		a = b
		b = c
	return b

c = int(sys.argv[1])
start = time()
for i in range(0,c):
	print (fib(i))
	print("")

end = time()
print ("this took " + str(end - start))

File: fib-iterate.cpp
#include <omp.h>
#include <stdlib.h>
#include <iostream>
	
typedef unsigned long long int mylongint;
mylongint fib(int x) {
	mylongint a = 1, b = 1;
	for (mylongint i = 3; i <= x; i++) {
		mylongint c = a + b;
		a = b;
		b = c;
	}
	return b;
}

int main(int argc, char ** argv) {

	double start = omp_get_wtime();
	unsigned int n = atoi(argv[1]);
	for (unsigned int i = 0; i < n; i++) {
		std::cout << fib(i) << "\n" << std::endl;
	}
	double end = omp_get_wtime();
	double diff = end-start;
	std::cout << "This took " << diff << " seconds to execute!";
}
